src/common/ui/ProgressWindow.jsx

```diff
@@ -127,7 +127,7 @@ Zotero.UI.ProgressWindow = class ProgressWindow extends React.PureComponent {
 
 		// Check if the X button to cancel a saving process should be displayed
 		(async () => {
-			let notSavingToWebLibrary = await Zotero.Connector.getClientVersion();
+			let notSavingToWebLibrary = await Zotero.Connector.checkIsOnline();
```

Comment:
This issues a superfluous `/ping` call to the client. The ProgressWindow should already know whether it's saving to Zotero or the server.

Reply:
It looks like we always try to [get selected collections](https://github.com/zotero/zotero-connectors/blob/ea7878807658fa73dbf82976ce00c8431129c583/src/common/inject/progressWindow_inject.js#L105) from the client (even if there is no zotero client running) and pass the targets (which are empty in case of saving to web) to `changeHeadline` which updates the state of ProgressWindow. And then, `this.state.targets` is used to [determine if the headline select](https://github.com/zotero/zotero-connectors/blob/ea7878807658fa73dbf82976ce00c8431129c583/src/common/ui/ProgressWindow.jsx#L595) for collections should be rendered. Is`this.state.targets` a reasonable way to determine if we are saving to desktop vs web? It does seems to work for instances that I tried, I just wasn't sure if there are times when it would not be the case.

Reply:
I think it's just `this.state.targetSelectorShown`? But target selector is written by Dan.

Reply:
`targetSelectorShown` is [updated](https://github.com/zotero/zotero-connectors/blob/ea7878807658fa73dbf82976ce00c8431129c583/src/common/ui/ProgressWindow.jsx#L444) each time when the chevron is clicked, so I think it is about whether the target selector table and tags region is expanded or collapsed...

But judging by [this comment as well](https://github.com/zotero/zotero-connectors/blob/ea7878807658fa73dbf82976ce00c8431129c583/src/common/ui/ProgressWindow.jsx#L198), does it seem like the existence of `targets` should imply that we fetched them from the client, while their non-existence should imply that we'll save to the server? I feel like it does... Combined with the fact that in `updateFromClient`, "Saving to zotero.org" is set without any targets and "Saving to ..." is set with the targets, I am somewhat convinced that `this.state.targets` == saving to the client.

Reply:
@dstillman, correct me if I am wrong on the above though!